### PR TITLE
Stomp planner

### PR DESCRIPTION
**Pull request by bbrito**
Stomp planner configuration added for raw3-1. Launch files too.

merge together with https://github.com/ipa320/cob_manipulation/pull/104

### CODE REVIEW

**Review by floweisshardt — DISMISSED** [review]
no launch files in cob_hardware https://github.com/ipa320/cob_robots/pull/631/files#diff-22447eacd575bff1ed87005f7ac42b32 (this is also in cob4-5 directory which is certainly not correct.)

**Comment by bbrito** [comment]
Which launch files are you referring to?

**Comment by floweisshardt** [comment]
renamed/moved one (probably not by purpose): `cob_bringup/components/legacy_schunk_powercube.launch → cob_hardware_config/cob4-5/config/legacy_schunk_powercube.launch`

File: cob_bringup/robots/cob4-5.xml [hunk 1]
```diff
@@ -131,6 +132,7 @@
 			<arg name="component_name" value="arm_left"/>
 			<arg name="can_device" value="can3"/>
 			<arg name="sim" value="$(arg sim)"/>
+			<arg name="cartesian_control" value="$(arg cartesian_control)"/>
```
Comment (fmessmer) [inline cob_bringup/robots/cob4-5.xml]: if we add this, then we add it for both arms...
however, adding it currently gives us load problems on the real hardware...
you could pass the argument only in case sim is active:
`<arg if=$(arg sim) name="cartesian_control" value="$(arg cartesian_control)"/>`

File: cob_bringup/robots/cob4-5.xml [hunk 2]
```diff
@@ -11,6 +11,7 @@
 	<arg name="cob4-5-h1" default="localhost"/>
 	<arg name="env-script" default="$(find cob_bringup)/env.sh"/>
 	<arg name="sim" default="false"/>
+	<arg name="cartesian_control" default="false"/>
```
Comment (fmessmer) [inline cob_bringup/robots/cob4-5.xml]: remove this as you cannot call a xml directly thus you cannot pass a launch argument..

File: cob_bringup/robots/raw3-1.xml [hunk 3]
```diff
@@ -58,7 +59,7 @@
 			<arg name="robot" value="$(arg robot)"/>
 			<arg name="sim" value="$(arg sim)"/>
 		</include>
-		<include file="$(find cob_bringup)/components/legacy_schunk_powercube.launch">
+		<include file="$(find cob_bringup)/components/canopen_generic.launch">
```
Comment (fmessmer) [inline cob_bringup/robots/raw3-1.xml]: this won't work without proper configuration/yaml files...I guess this hasn't been tested on HW...

File: cob_hardware_config/raw3-1/config/arm_controller.yaml [hunk 4]
```diff
@@ -32,4 +32,78 @@ joint_trajectory_controller:
...
+  joints:
+     - arm_shoulder_pan_joint
+     - arm_shoulder_lift_joint
+     - arm_elbow_joint
+     - arm_wrist_1_joint
+     - arm_wrist_2_joint
+     - arm_wrist_3_joint
+  required_drive_mode: 1
+
+arm_1_joint_position_controller:
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_controller.yaml]: convention is to use <joint_name>_position_controller....-> arm_shoulder_pan_joint_position_controller
same for the other controllers...

File: cob_hardware_config/raw3-1/config/arm_controller.yaml [hunk 5]
```diff
@@ -32,4 +32,78 @@ joint_trajectory_controller:
...
+     - arm_shoulder_pan_joint
+     - arm_shoulder_lift_joint
+     - arm_elbow_joint
+     - arm_wrist_1_joint
+     - arm_wrist_2_joint
+     - arm_wrist_3_joint
+  required_drive_mode: 2
+
+arm_1_joint_velocity_controller:
+  type: velocity_controllers/JointVelocityController
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_controller.yaml]: have you tested this on hardware?
does the ur_modern_driver support velocity interfaces? if yes, then please use ur_modern_driver in the ur.launch....with proper configuration....so far, this seems to be simulation only

**Review by fmessmer — CHANGES_REQUESTED** [review]
Please review your PR for formal correctness before we can review the actual feature:
 - do not change indentation scheme
 - revert moving files
 - reference/link related PRs

File: cob_moveit_config/raw3-1/config/controllers.yaml [hunk 6]
```diff
@@ -9,9 +9,14 @@ controller_list:
...
       - arm_wrist_3_joint
-  - name: torso_controller
+  - name: torso/joint_trajectory_controller
     action_ns: follow_joint_trajectory
     type: FollowJointTrajectory
     joints:
       - torso_pan_joint
       - torso_tilt_joint
+  - name: multi_dof_joint_trajectory_action
+    type: MultiDofFollowJointTrajectory
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/controllers.yaml]: this won't work for others...that do not have this plugin

File: cob_moveit_config/raw3-1/config/base_controller.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+controller_list:
+  - name: base_controller
+    joints:
+      - virtual/x
+      - virtual/y
+      - virtual/theta
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/base_controller.yaml]: is this used by the MultiDofFollowJointTrajectory plugin? why not use base/x instead of virtual/x?

File: cob_moveit_config/raw3-1/config/fake_controllers.yaml [hunk 8]
```diff
@@ -6,8 +6,4 @@ controller_list:
       - arm_elbow_joint
       - arm_wrist_1_joint
       - arm_wrist_2_joint
-      - arm_wrist_3_joint
-  - name: fake_torso_controller
-    joints:
-      - torso_pan_joint
-      - torso_tilt_joint
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/fake_controllers.yaml]: keep the fake_torso_controller

File: cob_moveit_config/raw3-1/config/kinematics.yaml [hunk 9]
```diff
@@ -1,16 +1,5 @@
...
   kinematics_solver: kdl_kinematics_plugin/KDLKinematicsPlugin
-  #kinematics_solver: cob_kinematics/IKFast_ur10
   kinematics_solver_search_resolution: 0.005
-  kinematics_solver_timeout: 0.005
-  kinematics_solver_attempts: 3
-torso:
-  kinematics_solver: kdl_kinematics_plugin/KDLKinematicsPlugin
-  kinematics_solver_search_resolution: 0.005
-  kinematics_solver_timeout: 0.05
-  kinematics_solver_attempts: 3
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/kinematics.yaml]: keep the torso kinematics plugin

File: cob_moveit_config/raw3-1/config/ompl_planning.yaml [hunk 10]
```diff
@@ -26,14 +26,13 @@ planner_configs:
     min_valid_path_fraction: 0.5  # Accept partially valid moves above fraction. default: 0.5
   RRTkConfigDefault:
     type: geometric::RRT
-    range: 0.1  # Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()
+    range: 0.0  # Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/ompl_planning.yaml]: if this PR is about adding STOMP...why do you change OMPL parameters?

File: cob_moveit_config/raw3-1/config/raw3-1.srdf [hunk 11]
```diff
@@ -4,8 +4,16 @@
...
+    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
+    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
+    <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
+    <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
+    <group name="arm">
+        <chain base_link="base_link" tip_link="gripper" />
+    </group>
     <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
-    <virtual_joint name="world_joint" type="planar" parent_frame="odom_combined" child_link="base_footprint" />
+    <virtual_joint name="Base" type="planar" parent_frame="/odom_combined" child_link="base_footprint" />
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/raw3-1.srdf]: why change the name?

**Review by fmessmer — CHANGES_REQUESTED** [review]
few more...

File: cob_bringup/robots/raw3-1.xml (hunk 3 above)
Comment (bbrito) [reply cob_bringup/robots/raw3-1.xml]: The problem is that "someone" deleted legacy_schunk_powercube.launch so now the software that we have in the main repository is wrong fro raw3-1 and cob3-6. Anyway I will put this back as it was...

File: cob_hardware_config/raw3-1/config/arm_controller.yaml (hunk 5 above)
Comment (bbrito) [reply cob_hardware_config/raw3-1/config/arm_controller.yaml]: I tested it at it works. I will keep the both options for the drivers for the ones that don't have the new driver... The configuration files are inside the driver package. I don't think it is  a good idea to change that...

File: cob_moveit_config/raw3-1/config/base_controller.yaml (hunk 7 above)
Comment (bbrito) [reply cob_moveit_config/raw3-1/config/base_controller.yaml]: yes.

File: cob_bringup/components/legacy_schunk_powercube.launch [hunk 12]
```diff
@@ -20,6 +20,6 @@
 			<arg name="robot" value="$(arg robot)"/>
 			<arg name="component_name" value="$(arg component_name)"/>
 		</include>
-		<node ns="$(arg component_name)" pkg="cob_helper_tools" type="fake_driver.py" name="fake_driver" cwd="node" respawn="false" output="screen"/>
 	</group>
 </launch>
+
```
Comment (fmessmer) [inline cob_bringup/components/legacy_schunk_powercube.launch]: fully remove changes in this file...

File: cob_bringup/robots/cob4-5.xml [hunk 13]
```diff
@@ -12,6 +12,7 @@
 	<arg name="env-script" default="$(find cob_bringup)/env.sh"/>
 	<arg name="sim" default="false"/>
 
+
```
Comment (fmessmer) [inline cob_bringup/robots/cob4-5.xml]: remove

File: cob_bringup/robots/cob4-5.xml [hunk 14]
```diff
@@ -131,6 +132,7 @@
 			<arg name="component_name" value="arm_left"/>
 			<arg name="can_device" value="can3"/>
 			<arg name="sim" value="$(arg sim)"/>
+			<arg if="$(arg sim)" name="cartesian_control" value="$(arg cartesian_control)"/>
```
Comment (fmessmer) [inline cob_bringup/robots/cob4-5.xml]: there is no `$(arg cartesian_control`...set value to `true` in case you like to use it...

also add it to arm_right

File: cob_bringup/robots/raw3-1.xml [hunk 15]
```diff
@@ -68,6 +69,7 @@
 			<arg name="use_modern_driver" value="false"/>
 			<arg name="robot_ip" value="$(arg ur_ip)"/>
 			<arg name="sim" value="$(arg sim)"/>
+			<arg name="cartesian_control" value="$(arg cartesian_control)"/>
```
Comment (fmessmer) [inline cob_bringup/robots/raw3-1.xml]: make consistent with the comments for cob4-5

File: cob_hardware_config/cob4-5/config/legacy_schunk_powercube.launch [hunk 16]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
```
Comment (fmessmer) [inline cob_hardware_config/cob4-5/config/legacy_schunk_powercube.launch]: remove this file...it's in cob_bringup/components!

File: cob_hardware_config/raw3-1/config/arm_controller.yaml (hunk 4 above)
Comment (fmessmer) [reply cob_hardware_config/raw3-1/config/arm_controller.yaml]: please respect the convention!

File: cob_hardware_config/raw3-1/config/arm_controller.yaml (hunk 5 above)
Comment (fmessmer) [reply cob_hardware_config/raw3-1/config/arm_controller.yaml]: What did you test? On hardware?
In case it works with the (released) ur_modern_driver and in simulation, it's ok to use the ur_modern_driver as the default option in ur.launch...as long as the dependency is defined...

File: cob_moveit_config/raw3-1/config/kinematics.yaml [hunk 17]
```diff
@@ -1,16 +1,11 @@
 arm:
   kinematics_solver: kdl_kinematics_plugin/KDLKinematicsPlugin
-  #kinematics_solver: cob_kinematics/IKFast_ur10
   kinematics_solver_search_resolution: 0.005
-  kinematics_solver_timeout: 0.005
+  kinematics_solver_timeout: 5
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/kinematics.yaml:4]: 5 seconds for an IK solution?
You could use `ur_kinematics/UR10KinematicsPlugin` it's an analytic solver rather than the numerical KDL solver

File: cob_moveit_config/raw3-1/config/ompl_planning.yaml [hunk 18]
```diff
@@ -30,10 +30,9 @@ planner_configs:
     goal_bias: 0.05  # When close to goal select goal, with this probability? default: 0.05
   RRTConnectkConfigDefault:
     type: geometric::RRTConnect
-    range: 0.1  # Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()
+    range: 0.0  # Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()
   RRTstarkConfigDefault:
     type: geometric::RRTstar
-    optimization_objective: MaximizeMinClearanceObjective
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/ompl_planning.yaml:36]: why did you change these values?
do the previous not result in a satisfying behavior?

File: cob_moveit_config/raw3-1/config/raw3-1.srdf [hunk 19]
```diff
@@ -4,8 +4,16 @@
     A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
 -->
 <robot name="raw3-1">
+    <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
+    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
+    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
+    <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
+    <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
+    <group name="arm">
+        <chain base_link="base_link" tip_link="gripper" />
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/raw3-1.srdf]: is this group correct?
should it not be `arm_base_link` -> `arm_ee_link`?

**Review by fmessmer — CHANGES_REQUESTED** [review]
still:
 - restore indentation

and we have some dependency issues as the required packages are apparently not released into indigo

File: cob_hardware_config/raw3-1/config/arm_controller.yaml (hunk 5 above)
Comment (bbrito) [reply cob_hardware_config/raw3-1/config/arm_controller.yaml]: I will keep like this for now. If you guys want in the future to make it default feel free.

File: cob_moveit_config/raw3-1/config/raw3-1.srdf (hunk 11 above)
Comment (bbrito) [reply cob_moveit_config/raw3-1/config/raw3-1.srdf]: it is more representative...

File: cob_moveit_config/raw3-1/config/kinematics.yaml (hunk 17 above)
Comment (bbrito) [reply cob_moveit_config/raw3-1/config/kinematics.yaml:4]: This was the best configuration used... Trust me... I'm an engineer ;)

File: cob_moveit_config/raw3-1/config/ompl_planning.yaml (hunk 18 above)
Comment (bbrito) [reply cob_moveit_config/raw3-1/config/ompl_planning.yaml:36]: No.

File: cob_bringup/drivers/openni.launch [hunk 20]
```diff
@@ -3,7 +3,7 @@
 
 	<arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
 	<arg name="name" default="cam3d"/>
-	<arg name="device_id" default="#1"/>
+	<arg name="device_id" default="2@0"/>
```
Comment (fmessmer) [inline cob_bringup/drivers/openni.launch]: revert
call `openni.launch` from `raw3-1.xml` with proper launch argument instead

File: cob_bringup/drivers/ur.launch [hunk 21]
```diff
@@ -1,37 +1,43 @@
...
+    </group>
+    <include if="$(arg sim)" file="$(find cob_bringup)/controllers/generic_controller.launch">
+        <arg name="robot" value="$(arg robot)"/>
+        <arg name="component_name" value="$(arg component_name)"/>
+    </include>
+    <include if="$(arg cartesian_control)" file="$(find cob_bringup)/controllers/generic_cartesian_controller.launch">
+        <arg name="robot" value="$(arg robot)"/>
+        <arg name="component_name" value="$(arg component_name)"/>
+        <arg name="debug" value="false"/>
+    </include>
```
Comment (fmessmer) [inline cob_bringup/drivers/ur.launch]: tabs vs. spaces

File: cob_bringup/robots/cob4-5.xml [hunk 22]
```diff
@@ -131,6 +131,7 @@
 			<arg name="component_name" value="arm_left"/>
 			<arg name="can_device" value="can3"/>
 			<arg name="sim" value="$(arg sim)"/>
+			<arg if="$(arg sim)" name="cartesian_control" value="true"/>
```
Comment (fmessmer) [inline cob_bringup/robots/cob4-5.xml:134]: as long as we use the "old" `cob_obstacle_distance` we cannot use this on the real hardware because it leads to high load
however, I think `if="$(arg sim)"` is ok

but add it to to **both** arms!

File: cob_bringup/robots/raw3-1.xml [hunk 23]
```diff
@@ -9,6 +9,7 @@
 	<arg name="ur_ip" default="localhost"/>
 	<arg name="env-script" default="$(find cob_bringup)/env.sh"/>
 	<arg name="sim" default="false"/>
+	<arg name="cartesian_control" default="true"/>
```
Comment (fmessmer) [inline cob_bringup/robots/raw3-1.xml]: remove this line

File: cob_bringup/robots/raw3-1.xml [hunk 24]
```diff
@@ -68,6 +69,7 @@
 			<arg name="use_modern_driver" value="false"/>
 			<arg name="robot_ip" value="$(arg ur_ip)"/>
 			<arg name="sim" value="$(arg sim)"/>
+			<arg if="$(arg sim)" name="cartesian_control" value="true"/>
```
Comment (fmessmer) [inline cob_bringup/robots/raw3-1.xml:71]: ok, see https://github.com/ipa320/cob_robots/pull/631/files#r109871065

File: cob_hardware_config/raw3-1/config/arm_cartesian_controller.yaml [hunk 25]
```diff
@@ -0,0 +1,79 @@
+## Cartesian controller parameters
+chain_base_link: arm_base_link
+chain_tip_link: arm_ee_link
+root_frame: odom_combined
+
+# twist controller parameters
+twist_controller:
+  controller_interface: 0    #Velocity 0, Position 1, Trajectory 2, JointStates 3
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_cartesian_controller.yaml]: use the new plugin-based interface configuration

Comment (fmessmer) [reply cob_hardware_config/raw3-1/config/arm_cartesian_controller.yaml]: preferably you meld this file against any other cartesian_controller.yaml for consistency...

File: cob_hardware_config/raw3-1/config/arm_controller.yaml [hunk 26]
```diff
@@ -32,4 +32,78 @@ joint_trajectory_controller:
...
+joint_group_position_controller:
+  type: position_controllers/JointGroupPositionController
+  joints:
+     - arm_shoulder_pan_joint
+     - arm_shoulder_lift_joint
+     - arm_elbow_joint
+     - arm_wrist_1_joint
+     - arm_wrist_2_joint
+     - arm_wrist_3_joint
+  required_drive_mode: 1
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_controller.yaml]: required_drive_mode is `ros_canopen`-specific...remove!

File: cob_moveit_config/raw3-1/config/kinematics.yaml [hunk 27]
```diff
@@ -1,16 +1,11 @@
 arm:
   kinematics_solver: kdl_kinematics_plugin/KDLKinematicsPlugin
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/kinematics.yaml:2]: try the analytical, UR-specific `ur_kinematics/UR10KinematicsPlugin`

File: cob_moveit_config/raw3-1/config/raw3-1.srdf [hunk 28]
```diff
@@ -4,8 +4,16 @@
...
+    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
+    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
+    <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
+    <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
+    <group name="arm">
+        <chain base_link="arm_base_link" tip_link="arm_ee_link" />
+    </group>
     <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
-    <virtual_joint name="world_joint" type="planar" parent_frame="odom_combined" child_link="base_footprint" />
+    <virtual_joint name="base" type="planar" parent_frame="/odom_combined" child_link="base_footprint" />
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/raw3-1.srdf]: I guess you don't need the `/`

**Review by fmessmer — CHANGES_REQUESTED** [review]

File: cob_moveit_config/raw3-1/config/kinematics.yaml (hunk 27 above)
Comment (bbrito) [reply cob_moveit_config/raw3-1/config/kinematics.yaml:2]: Nor working for now:

Kinematics solver of type 'ur_kinematics/UR10KinematicsPlugin' could not be initialized for group 'arm'

We keep the previous solver.

File: cob_bringup/drivers/ur.launch (hunk 21 above)
Comment (fmessmer) [reply cob_bringup/drivers/ur.launch]: still tabs vs. spaces! please remove spaces and use tabs instead

File: cob_bringup/robots/cob4-5.xml [hunk 29]
```diff
@@ -1,255 +1,257 @@
 <?xml version="1.0"?>
 <launch>
```
Comment (fmessmer) [inline cob_bringup/robots/cob4-5.xml]: tabs vs. spaces

File: cob_hardware_config/raw3-1/config/arm_cartesian_controller.yaml [hunk 30]
```diff
@@ -0,0 +1,79 @@
...
+  activation_threshold_ca: 0.1    #in [m]
+  damping_ca: 0.000001
+
+  # limit enforcing
+  keep_direction: true
+  enforce_pos_limits: true
+  enforce_vel_limits: true
+  enforce_acc_limits: false
+  limits_tolerance: 5.0    #for position limits [deg]
+  #limits_acc: [10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0]    #[rad/s2]
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_cartesian_controller.yaml:47]: please update according to https://github.com/ipa320/cob_robots/blob/indigo_dev/cob_hardware_config/cob4-5/config/arm_left_cartesian_controller.yaml#L40-L49

File: cob_moveit_config/raw3-1/config/raw3-1.srdf [hunk 31]
```diff
@@ -4,8 +4,25 @@
...
 <robot name="raw3-1">
+    <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
+    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
+    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
+    <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
+    <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
+    <group name="arm">
+        <chain base_link="arm_base_link" tip_link="arm_ee_link" />
+    </group>
+    <group name="robot">
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/raw3-1.srdf]: when running `roslaunch cob_moveit_bringup demo.launch robot:=raw3-1`, I see
```
Group 'robot' does not have a parent link
```

File: cob_moveit_config/raw3-1/config/raw3-1.srdf [hunk 32]
```diff
@@ -4,8 +4,25 @@
     A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
 -->
 <robot name="raw3-1">
+    <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
+    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
+    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
+    <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
+    <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
+    <group name="arm">
+        <chain base_link="arm_base_link" tip_link="arm_ee_link" />
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/raw3-1.srdf:13]: also, there is no end-effector defined for `arm`, thus, you do not get an interactive marker at the to move the GoalState in rviz!

File: cob_moveit_config/raw3-1/config/kinematics.yaml (hunk 27 above)
Comment (fmessmer) [reply cob_moveit_config/raw3-1/config/kinematics.yaml:2]: the `ur_kinematics/UR10KinematicsPlugin` also takes a `arm_prefix` parameter...

**Review by fmessmer — CHANGES_REQUESTED** [review]

File: cob_bringup/robots/cob4-5.xml (hunk 29 above)
Comment (bbrito) [reply cob_bringup/robots/cob4-5.xml]: Now you want tabs??? I don't understand

**Comment by bbrito** [comment]
Also the official branch of ur arm does not use that plugin:

https://github.com/ros-industrial/universal_robot/blob/kinetic-devel/ur10_moveit_config/config/kinematics.yaml

File: cob_moveit_config/raw3-1/config/raw3-1.srdf (hunk 31 above)
Comment (bbrito) [reply cob_moveit_config/raw3-1/config/raw3-1.srdf]: Its normal because is creating a virtual joint to a link (odom_combined) that does not exist yet... after that is fine

File: cob_moveit_config/raw3-1/config/raw3-1.srdf (hunk 32 above)
Comment (bbrito) [reply cob_moveit_config/raw3-1/config/raw3-1.srdf:13]: I don't need that for the moment... THere was never a end-effector here xD

File: cob_bringup/robots/cob4-5.xml (hunk 29 above)
Comment (fmessmer) [reply cob_bringup/robots/cob4-5.xml]: it's always been tabs :wink:

**Review by fmessmer — APPROVED** [review]
